### PR TITLE
[GEN-450] Gym dropdown populated by gyms collection documents

### PR DESCRIPTION
**Pull request by redpepper-nag**
**Overview**
1. Creation of GymsApi with function that retrieves all the gyms documents and saves their names and ids
2. Using GymsApi to render a list of all the gyms documents (showing the name in the form dropdown and saving the id of the gym as the key value pair of gym in a newly created climb document)

**iOS**
IMG_512226FFC098-1
**Android**
371192394_1315118995860356_4249399432495428898_n

### CODE REVIEW

File: trk/src/Screens/ClimbCreate/index.js [hunk 1]
```diff
@@ -62,11 +63,14 @@ const ClimbInputData = (props) => {
...
-    { label: 'VITAL', value: 'VITAL' },
-    { label: 'The Gravity Vault Hoboken', value: 'The Gravity Vault Hoboken' }]);
+  const [gymItems, setGymItems] = useState([]);
+
+  useEffect(() => {
+    // Fetch gyms when component mounts
+    GymsApi().fetchGyms().then(gyms => {
+      setGymItems(gyms);
+    });
+  }, []);
```
Comment (redpepper-nag) [inline trk/src/Screens/ClimbCreate/index.js:73]: Using the fetchGyms function from GymsApi to access an array of all the gyms documents containing the name and id of each gym). These gyms set the gym state

File: trk/src/Screens/ClimbCreate/index.js [hunk 2]
```diff
@@ -303,6 +310,7 @@ const ClimbInputData = (props) => {
                 placeholderTextColor={"#b1b1b3"}
                 onChangeText={setInfo}
                 placeholder="Enter more info"
+                multiline={true}
```
Comment (redpepper-nag) [inline trk/src/Screens/ClimbCreate/index.js:313]: this is to fix the fact that the Climb info input text did not wrap onto the next line

File: trk/src/Screens/ClimbCreate/index.js [hunk 3]
```diff
@@ -381,7 +389,7 @@ const styles = StyleSheet.create({
     fontSize: 18,
     color: 'black',
     borderRadius: 8,
-    paddingBottom: 120,
+    paddingBottom: 20,
     paddingLeft: 10,
```
Comment (redpepper-nag) [inline trk/src/Screens/ClimbCreate/index.js:393]: had to reduce the bottom padding to allow the wrapping of text in climb info

File: trk/src/api/GymsApi.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    }
+  }
+
+  return {
+    fetchGyms,
+  }
+}
+
+export default GymsApi
+
```
Comment (redpepper-nag) [inline trk/src/api/GymsApi.js:22]: this function fetches all the gym documents in the gyms collection, maps each gym and sets the label as the name of the gym and the id as the value of each gym

File: trk/src/api/GymsApi.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+function GymsApi() {
+
+  async function fetchGyms() {
+    try {
+      const snapshot = await firestore().collection('gyms').get();
+      return snapshot.docs.map(doc => ({ label: doc.data().Name, value: doc.id }));
+    } catch (error) {
+      console.error("Error fetching gyms: ", error);
+      return [];
+    }
```
[the remark this answers is not in the document]
Comment (VGmt) [reply trk/src/api/GymsApi.js:13]: Aussi, habituellement, à part si je me trompe, on fait la data manipulation coté front-end:
`return snapshot.docs.map(doc => ({ label: doc.data().Name, value: doc.id }));`

On peut garder - idk what's the best practice for API calls. Only pointing it out as we're usually not doing it this way

**Review by VGmt — CHANGES_REQUESTED** [review]

File: trk/src/api/GymsApi.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (redpepper-nag) [reply trk/src/api/GymsApi.js:13]: you're right - so doing this part directly in create climb?

[the remark this answers is not in the document]
Comment (VGmt) [reply trk/src/api/GymsApi.js:13]: Thing is, I would understand why a function called **FetchGym** does the data manipulation on the backend. It's supposed to present you with a list of gyms. It should do what it's called.

IMO keep the **mapping** there.

**Review by VGmt — APPROVED** [review]

**Pull request merged**